packages/glimmer-runtime/tests/ember-component-test.ts

```diff
@@ -871,6 +871,59 @@ testComponent('parameterized has-block-params (concatted attr, default) when blo
   expected: '<button data-has-block-params="is-false"></button>'
 });
 
+module("Components - curlies - dynamic component");
+
+QUnit.test('missing definition', assert => {
```

Comment:
Added tests for both cases. I'm not sure why it's rendering a comment. I can dig in and see. It makes sense that it puts _something_ in the DOM to keep track of where it goes, but I'm not sure why it's a comment.